lightningd/peer_control.c

```diff
@@ -1192,6 +1192,9 @@ static struct command_result *json_setchannelfee(struct command *cmd,
 	const jsmntok_t *idtok;
 	unsigned int *base;
 	unsigned int *ppm;
+	struct peer *peer;
+	/* FIXME: gcc 7.3.0 thinks this might not be initialized. */
```

Comment:
This should be easy to fix by returning in the `else` branch on line 1213 always.

Reply:
this was actually a copyNpaste from same file which contains same logic for function:
```C
static struct command_result *json_close(struct command *cmd, ...)
```

Always returning in else is imho not an option, as `command_find_channel` sets the pointer as fourth argument: `&channel`.  Or maybe I misunderstood your suggestion. I can remove the comment though, as initializing with NULL seems fine to me :)